### DISCUSSION TITLE
Purpose of `useSearchParams`?

### DISCUSSION THREAD

**Discussion by bestickley** [vercel/next.js#51048] [Help]
### Summary

What's the purpose of `useSearchParams`? Is it any different than a readonly version of `new URLSearchParams(location.search)`? Is it doing anything else special for Next.js behind the scenes?

**Comment by bestickley** [comment]
Ah I think the purpose is so that you can reference search params without having to use `location.search` which isn't available on server. However, this leads me to another question. Why are client components running on server? Are the pre-rendered? If so, shouldn't the hooks not be running?

**Reply by bestickley** [reply]
Found my answer [here](https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering#dynamic-functions)

> Using [useSearchParams()](https://nextjs.org/docs/app/api-reference/functions/use-params) in Client Components will skip static rendering and instead render all Client Components up to the nearest parent Suspense boundary on the client.
We recommend wrapping the Client Component that uses useSearchParams() in a <Suspense/> boundary. This will allow any Client Components above it to be statically rendered. [Example](https://nextjs.org/docs/app/api-reference/functions/use-search-params#static-rendering).

**Reply by bestickley** [reply]
But I'm still confused why if I use `useSearcParams` once in my client component and I still use `location.search` elsewhere, I get an error when running dev server (turbopack).

**Reply by tinleym** [reply]
Using useSearchParams in a client event callback function, I wasn't always getting the current search params from the URL. Simply using location.search gives me the right result every time. 

I also don't understand what this hook is for or when it should be used.

Update: location.search causes issues when rendered on the server... still, useSearchParams doesn't necessarily update when the url does. So I have components that use both... useSearchParams for initialization, and location.search for updates in event callbacks.
